core/src/main/java/com/yahoo/oak/Block.java

```diff
@@ -6,15 +6,11 @@
 
 package com.yahoo.oak;
 
-import sun.misc.Cleaner;
-
-import java.lang.reflect.Field;
-import java.nio.ByteBuffer;
 import java.util.concurrent.atomic.AtomicLong;
 
 class Block {
 
-    private final ByteBuffer buffer;
+    private final long memAddress;
```

Comment:
Please add some comment about what this address is, and explain the ability of having a ByteBuffer here.

Reply:
I would name it `blockMemAddress` as the start of the big chunk of memory related to the Block. And inside the Slice I would add Slice's offset and keep `sliceMemAddress`.